acceptancetests/deploy_stack.py

```diff
@@ -736,6 +771,32 @@ def from_args(cls, args):
         return cls.from_client(args, client)
 
     @classmethod
+    def from_existing(cls, args):
+        try:
+            juju_home = os.environ['JUJU_HOME']
+        except KeyError:
+            logging.error(
+                'Use of the --existing flag requires setting the JUJU_HOME '
+                'environmental variable. Please point JUJU_HOME to your local '
+                'juju data directory')
+        env = args.temp_env_name.split('-temp-env')[0]
```

Comment:
I believe this is to remove the default generated name? I feel this is a little clearer in intent (i.e. remove the trailing cruft):
args.temp_env_name.replace('-temp-env', '')